### ISSUE TITLE
Attempted to get trait from destroyed object (garefn 817 (not in world))

### ISSUE THREAD

**Issue by johannessorheim** [OpenRA/OpenRA#16873]
### Issue Summary
The game crashes during skirmish game

#### System Information
- Mac OS 10.14.6
- Mono JIT compiler version 5.16.0.220 (2018-06/bb3ae37d71a Fri Nov 16 17:12:11 EST 2018)
- release-20190314
- Red Alert 2, {DEV_VERSION}, master, revision b9668d9c9b1c6476

Platform is OSX
Engine version is release-20190314
Using SDL 2 with OpenGL renderer
Desktop resolution: 1280x800
No custom resolution provided, using desktop resolution
Using resolution: 1280x800
Using window scale 2,00
OpenGL version: 2.1 INTEL-12.10.12
Using default sound device
Internal mods:
	modcontent: Mod Content Manager (release-20190314)
	ra2: Red Alert 2 ({DEV_VERSION})
	all: All mods (release-20190314)
	cnc: Tiberian Dawn (release-20190314)
	d2k: Dune 2000 (release-20190314)
	ra: Red Alert (release-20190314)
	ts: Tiberian Sun (release-20190314)
External mods:
	ra-release-20190314: Red Alert (release-20190314)
	ra2-{DEV_VERSION}: Red Alert 2 ({DEV_VERSION})

2019-08-03T20:46:51] Game started
Exception of type `System.InvalidOperationException`: Attempted to get trait from destroyed object (garefn 817 (not in world))
  at OpenRA.TraitDictionary.CheckDestroyed (OpenRA.Actor actor) [0x0001f] in <7d8c6a66e78a4946b4adaa51d50f7df7>:0 

#### Additional Information:
- Steps to reproduce
	1. Play skirmish game
	2. Wait for it to crash

- Logs
[logs.zip](https://github.com/OpenRA/OpenRA/files/3464261/logs.zip)

- OpenRA Replays
The issue appears at 25:13, when my base is under attack.
[OpenRA-2019-08-03T184651Z.orarep.zip](https://github.com/OpenRA/OpenRA/files/3464280/OpenRA-2019-08-03T184651Z.orarep.zip)

**Comment by reaperrr** [comment]
The important part:

```
Exception of type `System.InvalidOperationException`: Attempted to get trait from destroyed object (garefn 817 (not in world))
  at OpenRA.TraitDictionary.CheckDestroyed (OpenRA.Actor actor) [0x0001f] in <7d8c6a66e78a4946b4adaa51d50f7df7>:0 
  at OpenRA.TraitDictionary.Get[T] (OpenRA.Actor actor) [0x00001] in <7d8c6a66e78a4946b4adaa51d50f7df7>:0 
  at OpenRA.Actor.Trait[T] () [0x00001] in <7d8c6a66e78a4946b4adaa51d50f7df7>:0 
  at OpenRA.Mods.Common.Activities.FindResources.GetSearchFromLocation (OpenRA.Actor self) [0x0004c] in <3c8e786c99d548118c2257eb76a5cf56>:0 
  at OpenRA.Mods.Common.Activities.FindResources.ClosestHarvestablePos (OpenRA.Actor self) [0x0006d] in <3c8e786c99d548118c2257eb76a5cf56>:0 
  at OpenRA.Mods.Common.Activities.FindResources.Tick (OpenRA.Actor self) [0x00043] in <3c8e786c99d548118c2257eb76a5cf56>:0 
```

**Comment by abcdefg30** [comment]
Hi and thanks for the report! This crash should be fixed by #16387 for the latest version.
